lib/features/children/details/repositories/child_details_repository.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:givt_app/core/network/api_service.dart';
+import 'package:givt_app/features/children/details/models/profile_ext.dart';
+import 'package:givt_app/features/children/overview/models/profile.dart';
+
+mixin ChildDetailsRepository {
+  Future<ProfileExt> fetchChildDetails(Profile profile);
+}
+
+class ChildDetailsRepositoryImpl with ChildDetailsRepository {
+  ChildDetailsRepositoryImpl(this.apiService);
+  final APIService apiService;
+
+  @override
+  Future<ProfileExt> fetchChildDetails(Profile profile) async {
```

Comment:
Why do you do the model operation in the repository? Which should only fetch and return the data? This breaks the single responsibility rule from SOLID principles. what do you think? I would do this in the cubit/bloc (as that should be the place for business operations)

Reply:
afaik repository can and should convert raw data (from provider) to model and then return it to cubit. Or I missed something?